bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/command/helper/ComponentStageHelper.java

```diff
@@ -0,0 +1,193 @@
+                    if (!StackUtils.isClientComponent(componentName)) {
+                        stageContext = createStageContext(componentName, hostnames, commandDTO);
+                        stages.add(new ComponentPrepareStage(stageContext));
+                    }
+                    break;
+            }
+        }
+
+        return stages;
+    }
+
+    private static StageContext createStageContext(
+            String componentName, List<String> hostnames, CommandDTO commandDTO) {
+        StageContext stageContext = StageContext.fromCommandDTO(commandDTO);
+
+        ServiceDTO serviceDTO = StackUtils.getServiceDTOByComponentName(componentName);
+        ComponentDTO componentDTO = StackUtils.getComponentDTO(componentName);
+
+        stageContext.setHostnames(hostnames);
+        stageContext.setServiceDTO(serviceDTO);
+        stageContext.setComponentDTO(componentDTO);
+
+        return stageContext;
+    }
+
+    private static List<String> getTodoList(List<String> componentNames, Command command) {
+        return getTodoList(componentNames, List.of(command));
+    }
+
+    private static List<String> getTodoList(List<String> componentNames, List<Command> commands) {
+        try {
+            List<String> orderedList =
+                    StackUtils.DAG.getAllNodesList().isEmpty() ? new ArrayList<>() : StackUtils.DAG.topologicalSort();
+            orderedList.replaceAll(String::toUpperCase);
+            List<String> componentCommandNames = new ArrayList<>();
+            for (String componentName : componentNames) {
+                for (Command command : commands) {
+                    String name =
+                            componentName.toUpperCase() + "-" + command.name().toUpperCase();
+                    componentCommandNames.add(name);
+                }
+            }
+
+            // Re-order the commands, since order.json does not contain all commands,
+            // only contains which has dependencies, we need to add the rest to the end.
+            if (commands.size() == 1) {
+                orderedList.retainAll(componentCommandNames);
+                componentCommandNames.removeAll(orderedList);
+                orderedList.addAll(componentCommandNames);
+                return orderedList;
+            } else {
+                // TODO, order.json currently only contains start/stop dependencies of major components, the situation
+                // of commands size greater than 1 is only with init/start/prepare commands, we see this as a special
+                // situation and use special logic for it, should use a better solution in the future.
+                orderedList.retainAll(componentCommandNames);
+                componentCommandNames.removeAll(orderedList);
+
+                List<String> res = new ArrayList<>();
+                List<String> commandOrder = List.of("ADD", "CONFIGURE", "INIT", "START", "PREPARE", "CHECK");
```

Comment:
Can an enumeration list be used here?

Reply:
This requires a lot of changes since it's not exactly the same as the Command enum class, I add a TODO for this code, we should optimize it in the future

Reply:
ok